### PR TITLE
Allow custom decoders for spec fields

### PR DESCRIPTION
**Pull request by hunner**
In nebula-sdk the decoder was a json decoder, and spec field receiving structs could implement UnmarshalJSON for custom decoding. The shift to mapstructure lost this ability. Luckily mapstructure 1.4.1 adds a TextUnmalshallerHookFunc that will apply when decoding fields that implement UnmarshalText.

Specifically: https://github.com/relay-integrations/relay-email/blob/d418e3930a1cad5d547653999f680e9377d2d95a/steps/email-sender-smtp/pkg/sender/spec.go#L8-L20

### CODE REVIEW

File: pkg/taskutil/spec.go [hunk 1]
```diff
@@ -119,6 +119,16 @@ func MetadataSpecURL() (string, error) {
 	return "", nil
 }
 
+type DefaultJSONSpecDecoder struct{}
+
+func (u DefaultJSONSpecDecoder) DecodeSpec(r io.Reader, v interface{}) error {
```
Comment (impl) [inline pkg/taskutil/spec.go:124]: Is this used anywhere? Seems like it isn't needed as it just directly wraps the JSON library?

**Review by impl — APPROVED** [review]
LGTM

**Comment by impl** [comment]
Don't forget to update CHANGELOG.md before you cut a new release.

File: pkg/taskutil/spec.go (hunk 1 above)
Comment (hunner) [reply pkg/taskutil/spec.go:124]: Ah heh, this was leftover from trying to add my own before I found the upstream change.

**Pull request merged**